syntax = "proto3";

package sentry_protos.billing.v1;

import "google/protobuf/timestamp.proto";
import "sentry_protos/billing/v1/data_category.proto";

// Aggregated usage counts for a single data category.
message UsageData {
  uint64 total = 1;
  uint64 accepted = 2;
  uint64 dropped = 3;
  uint64 filtered = 4;
  uint64 over_quota = 5;
  uint64 smart_limit = 6;
  uint64 dynamic_sampling = 7;
}

message GetUsageRequest {
  uint64 organization_id = 1;
  google.protobuf.Timestamp start = 2;
  google.protobuf.Timestamp end = 3;
}

message CategoryUsage {
  DataCategory category = 1;
  UsageData data = 2;
}

// Usage data for a single day.
message DailyUsage {
  google.protobuf.Timestamp date = 1;
  repeated CategoryUsage usage = 2;
}

message GetUsageResponse {
  // Usage broken down by day, each containing per-category usage.
  repeated DailyUsage days = 1;
}

service UsageService {
  rpc GetUsage(GetUsageRequest) returns (GetUsageResponse);
}
